### DISCUSSION TITLE
[Question]: How does function calling works ? [ Newbie ]

### DISCUSSION THREAD

**Discussion by drupalshift** [danny-avila/LibreChat#2902] [Q&A] [answered]
### What is your question?

How does function calling work in Assistants using LibreChat, I know how to use assistants in code, but where do I define my function ?

### More Details

For example, I have this function in my assistant 

```

[
  {
    "name": "get_product_info",
    "description": "Retrieve information for a given product by its ID",
    "parameters": {
      "type": "object",
      "properties": {
        "product_id": {
          "type": "string",
          "description": "ID of the product"
        }
      },
      "required": ["product_id"]
    }
  }
]

```

I know how to use assistants API directly using Python and impediment my function to retrieve products from an external API but how do I do that in LibreChat?

### What is the main subject of your question?

OpenAI Assistants Function Calling

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
at the moment, custom function handling is not supported. this would involve custom code, and you'd have to dig into the codebase to do this, but I plan on integrating a modular system for this.

For now, you can use openapi specs with actions.

**Reply by VML-Kai** [reply]
Any news on the modular system for custom function handling? Is this still a planned feature?

**Reply by danny-avila** [reply]
Yes, as time allows
